### ISSUE TITLE
Base default.nix and mavenix.nix does not accurately capture remotes

### ISSUE THREAD

**Issue by RocketPuppy** [icetan/mavenix#9]
This is workaroundable.

I have a project and I run the following steps:

1. mvnix-init
2. mvnix-update
3. nix-build ./default.nix -A build

On step three I receive this output:
```
$ nix-build ./default.nix -A build
building '/nix/store/phis5z67cspxrbfzl56b0lx0sqr1r2h4-remotes.json.drv'...
unpacking sources
unpacking source archive /nix/store/p17w448psv1s66qdf24gz3y9s57a94dj-converters
source root is converters
installing
error: fetchurl requires either `url` or `urls` to be set
(use '--show-trace' to show detailed location information)

$ nix-build ./default.nix -A build --show-trace
error: while evaluating the attribute 'buildPhase' of the derivation 'converters-1.0-SNAPSHOT' at /nix/store/c29dwdkc4pqb0gsg7nb3waqpk0z6brbv-nixos-18.03.132083.06c576b0525/nixos/pkgs/stdenv/generic/make-derivation.nix:148:11:
while evaluating the attribute 'buildCommand' of the derivation 'mk-repo' at /nix/store/c29dwdkc4pqb0gsg7nb3waqpk0z6brbv-nixos-18.03.132083.06c576b0525/nixos/pkgs/stdenv/generic/make-derivation.nix:148:11:
while evaluating 'urlToScript' at /home/dwilson/candidate_transparency/converters/mavenix.nix:19:27, called from undefined position:
while evaluating the derivation attribute 'name' at /nix/store/c29dwdkc4pqb0gsg7nb3waqpk0z6brbv-nixos-18.03.132083.06c576b0525/nixos/pkgs/stdenv/generic/make-derivation.nix:148:11:
fetchurl requires either `url` or `urls` to be set

```

I dug into the mavenix.nix flie and ended up running this command from the `getRemotes` function:
```
mvn 2>&- -B -nsu --offline --settings "${settings}" \
            dependency:list-repositories \
          | sed -n 's/.* \(id\|url\)://p' | tr -d '\n'
```
It gives me the remotes for my project (just central). If I add them to default.nix manually everything builds successfully. E.g.
```
{ pkgs ? import ../nix/nixpkgs.nix {}
, mavenix ? pkgs.callPackage (import ./mavenix.nix) {}
, src ? ./.
, doCheck ? false
}: mavenix {
  inherit src doCheck;
  infoFile = ./mavenix-info.json;
  remotes = {
    central = "https://repo.maven.apache.org/maven2";
  };

```

I'm not really sure what the problem is but the getRemotes function is generating an empty JSON object.

**Comment by RocketPuppy** [comment]
I extracted the getRemotes function into its own nix file and poked at it and I came across a potential lead but I'm stuck at the moment. I'm not sure of an easy way to debug it. The maven command to get remotes seemed to be failing for lack of a local repository. I attempted to use the mkRepo function to generate a local repo and pass it to getRemotes, but I ran into an infinite recursion issue.

Note I'm not entirely convinced that's a productive lead anyway.

**Comment by icetan** [comment]
I have encountered this before. It's the reason I added the `remotes` attribute as a workaround.

I also reached the conclusion that there might be missing dependencies when trying to get remotes. Probably missing parent POMs.

This hasn't been a problem in the projects I've been testing with, but they have a custom `settings.xml` which sets remotes.

I have a branch in progress that tries to put all POMs in the `mavenix-info.json`, not sure if this is a good way to go though. Feel free to test the `offline-parent-pom` branch.
